FastSimulation/Calorimetry/interface/CalorimetryManager.h

```diff
@@ -186,5 +188,8 @@ class CalorimetryManager{
   bool useShowerLibrary;
   bool useCorrectionSL;
   FastHFShowerLibrary *theHFShowerLibrary;
+
+  KKCorrectionFactors* ecalCorrection;
```

Comment:
use no dumb pointer here.
alternative
- no pointer
- std::unique_ptr

dumb pointers are error prone.
E.g. it seems you forgot to initialise it to 0 in the constructor of CalorimetryManager.